lib/components/user/monitored-trip/trip-status.js

```diff
@@ -20,9 +21,25 @@ const StyledFooterButton = styled(Button)`
 `
 
 export function FooterButton({ bsStyle, glyphIcon, onClick, text }) {
+  const [loading, setLoading] = useState(false)
+  const clickHandler = () => {
+    setLoading(true)
```

Comment:
Check that the action is not already pending before calling `onClick`.

Reply:
Have resolved this by adding `disabled` to the button.